### PR TITLE
feat(gh_112) - BaseVisualCamelEntity for Pipe

### PR DESCRIPTION
**Pull request by tplevko**
relates to - https://github.com/KaotoIO/kaoto-next/issues/112

### CODE REVIEW

File: packages/ui/src/models/camel-entities/flows/kamelet-binding.ts [hunk 1]
```diff
@@ -2,11 +2,16 @@ import { KameletBinding as KameletBindingModel } from '@kaoto-next/camel-catalog
...
+  KameletBindingSink,
+  KameletBindingSource,
+  KameletBindingStep,
+  KameletBindingSteps,
+} from '../kamelet-binding-overrides';
 
 export class KameletBinding implements BaseVisualCamelEntity {
-  readonly id = uuidv4();
-  readonly type = EntityType.KameletBinding;
+  id = uuidv4();
```
Comment (lordrip) [inline packages/ui/src/models/camel-entities/flows/kamelet-binding.ts:13]: [curious-question] Do we need to remove the `readonly` modifier?

File: packages/ui/src/models/camel-entities/flows/pipe.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Pipe as PipeModel } from '@kaoto-next/camel-catalog/types';
+import { v4 as uuidv4 } from 'uuid';
+import { BaseVisualCamelEntity, EntityType } from '../base-entity';
+import { KameletBinding } from '.';
```
Comment (lordrip) [inline packages/ui/src/models/camel-entities/flows/pipe.ts:4]:
```suggestion
import { KameletBinding } from './kamelet-binding';
```

File: packages/ui/src/models/camel-entities/flows/kamelet-binding.ts (hunk 1 above)
Comment (tplevko) [reply packages/ui/src/models/camel-entities/flows/kamelet-binding.ts]: with `readonly` for property `type`, there was following error in the Pipe class :

Property 'type' in type 'Pipe' is not assignable to the same property in base type 'KameletBinding'.

Comment (lordrip) [reply packages/ui/src/models/camel-entities/flows/kamelet-binding.ts]: Oh, interesting... No worries, I'll take a look later then 👍

**Review by lordrip — APPROVED** [review]

**Review by lordrip — APPROVED** [review]
LGTM @tplevko , thanks for the contribution 🎉

**Comment by lhein** [comment]
fixes #112

**Pull request merged**